reconfiguration/src/reconfiguration_handler.cpp

```diff
@@ -199,21 +199,26 @@ bool ReconfigurationHandler::handle(const concord::messages::GetDbCheckpointInfo
                                     uint32_t,
                                     const std::optional<bftEngine::Timestamp>&,
                                     concord::messages::ReconfigurationResponse& rres) {
-  concord::messages::GetDbCheckpointInfoStatusResponse response;
-  const auto& dbCheckpointList = DbCheckpointManager::instance().getListOfDbCheckpoints();
-  for (const auto& kv : dbCheckpointList) {
-    concord::messages::DbCheckpointInfo dbcpinfo_msg;
-    dbcpinfo_msg.seq_num = kv.second.lastDbCheckpointSeqNum_;
-    dbcpinfo_msg.block_id = kv.second.lastBlockId_;
-    dbcpinfo_msg.timestamp = kv.second.creationTimeSinceEpoch_.count();
+  if (bftEngine::ReplicaConfig::instance().dbCheckpointFeatureEnabled) {
+    concord::messages::GetDbCheckpointInfoStatusResponse response;
+    const auto& dbCheckpointList = DbCheckpointManager::instance().getListOfDbCheckpoints();
+    for (const auto& kv : dbCheckpointList) {
+      concord::messages::DbCheckpointInfo dbcpinfo_msg;
+      dbcpinfo_msg.seq_num = kv.second.lastDbCheckpointSeqNum_;
+      dbcpinfo_msg.block_id = kv.second.lastBlockId_;
+      dbcpinfo_msg.timestamp = kv.second.creationTimeSinceEpoch_.count();
 
-    response.db_checkpoint_info.push_back(dbcpinfo_msg);
-    LOG_INFO(getLogger(),
-             "GetDbCheckpointInfoStatus checkpoint id's are "
-                 << KVLOG(kv.second.lastBlockId_, kv.second.lastDbCheckpointSeqNum_));
+      response.db_checkpoint_info.push_back(dbcpinfo_msg);
+      LOG_INFO(getLogger(),
+               "GetDbCheckpointInfoStatus checkpoint id's are "
+                   << KVLOG(kv.second.lastBlockId_, kv.second.lastDbCheckpointSeqNum_));
+    }
+    rres.response = std::move(response);
+    return true;
+  } else {
+    LOG_WARN(getLogger(), "dbCheckpointFeature is disabled. GetDbCheckpointInfoRequest failed.");
+    return false;
```

Comment:
We can also set an error in the response. Something like:
`const auto err = "dbCheckpointFeature is disabled. GetDbCheckpointInfoRequest failed.";
    LOG_WARN(getLogger(), err);
    rres.response = concord::messages::ReconfigurationErrorMsg{err};`

Reply:
Fixed.